### PR TITLE
Return paths in flow group for diversity flow

### PR DESCRIPTION
**Pull request by IlyaMoiseev**
#1231 Return paths in flow group

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/1947)

### CODE REVIEW

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        int edgesOverlap = Sets.intersection(edges, mainFlowEdges).size();
+        int switchesOverlap = Sets.intersection(switches, mainFlowSwitches).size();
+        return new OverlappingSegmentsStats(edgesOverlap,
+                switchesOverlap,
+                percent(edgesOverlap, mainFlowEdges.size()),
+                percent(switchesOverlap, mainFlowSwitches.size()));
+    }
+
+    private int percent(int n, int from) {
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java]: what if it's zero?

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        return (int) ((n * 100.0f) / from);
+    }
+
+    /**
+     * Edge representation. Constraint: srcSwitch is always not less than destSwitch
+     */
+    @Value
+    static class Edge {
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java]: do we need nested class here?

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java (hunk 1 above)
Comment (IlyaMoiseev) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java]: For current logic from couldn't be zero. But if it is - floating point division will give NaN, and external cast to int gives zero. Added a test for this situation

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java (hunk 2 above)
Comment (IlyaMoiseev) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java]: This is highly specialized class for concrete purpose. And, as we already have several classes with that name, I preferred keep it as much restricted as possible.

**Comment by IlyaMoiseev** [comment]
@nikitamarchenko done

**Pull request merged**